pkg/cmd/pr/status/status_test.go

```diff
@@ -55,7 +56,12 @@ func runCommand(rt http.RoundTripper, branch string, isTTY bool, cli string) (*t
 		GitClient: &git.Client{GitPath: "some/path/git"},
 	}
 
-	cmd := NewCmdStatus(factory, nil)
+	withAllFeaturesDisabled := func(opts *StatusOptions) error {
```

Comment:
@vilmibm To set some context for our discussion tomorrow, this is the kind of test I want to talk about.

Since the `pr status` command now issues a detection query, and [we didn't want to mock `graphql` introspection in every test](https://github.com/cli/cli/pull/7462#discussion_r1196891663), I [found a rarely used pattern for injecting opts](https://github.com/cli/cli/blob/c90b87bdad40053adfddeac8dfc97446914cd3b2/pkg/cmd/issue/create/create_test.go#L379) but I noticed this was under a comment of `LEGACY TESTS`.

I have seen that there are some suites that:
 * Assert on the stdio of invoking `cmd.ExecuteC`
 * Assert on the result of applying `NewCmdFoo` to the `opts`, and assert on the stdio of invoking the private `fooRun` functions.
 
 I _suspect_ that the latter is the more recent way since it allows for testing with easy injection of `opts` and seems a little clearer than making use of the early return `runF` that calls `fooRun`. Let me know, cheers.

Reply:
The latter is more recent, yup. The thing that is legacy in the tests you linked is specifically the approach of making a `runCommand` helper then calling it in a series of top level test cases instead of using the list of test structs approach.